init-tools.sh

```diff
@@ -174,7 +174,10 @@ echo "Making all .sh files executable under Tools."
 ls $__scriptpath/Tools/*.sh | xargs chmod +x
 ls $__scriptpath/Tools/scripts/docker/*.sh | xargs chmod +x
 
+__compilersVersion=`ls $__PACKAGES_DIR/microsoft.netcore.compilers/ | sed 'r/\([0-9]\+\).*/\1/g'`
```

Comment:
Would we ever use a pre-release version of `microsoft.netcore.compilers`?  Does this regex account for pre-release versions?

Would there ever be more than 1 version here?  Could/should we just crossgen every version in this directory?

Reply:
We are currently using a prerelease version of microsoft.netcore.compilers, and this regex does work for it. I took this regex from the same one we use to detect the version of Microsoft.NETCore.App that comes with the CLI.

It is possible to have more than one version as you suggest, in the case where people don't clean their enlistment and sync their repos with different branches. I like your idea of doing it for each directory.